tests/functional/arxiv/test_arxiv.py

```diff
@@ -122,8 +123,5 @@ def test_arxiv(
         override_generated_fields(expected) for expected in expected_results
     ]
 
-    gotten_results = deep_sort(gotten_results)
-    expected_results = deep_sort(expected_results)
-
-    assert gotten_results == expected_results
+    assert DeepDiff(gotten_results, expected_results, ignore_order=True) == {}
```

Comment:
I'm not sure whether this is actually what we want, as ordering can be meaningful. The correct thing here is probably to do a shallow sort: we don't care in which order the records arrive (I see no reason why that would be non-deterministic in a test environment though), but we do care that the records are really equal, not equal up to arbitrary permutations, of say, the authors on the paper, as that's a sign of bugs :cry: not something I expect you to fix in this PR though.

Reply:
mmh, good point. I assumed that `deep_sort` was done with purpose and therefore simply looked at an equivalent way of comparing records. I do get different test results depending on what venv and what python version I run in, so the field order is not stable. certainly author lists should remain in stable sort order, but what about e.g. keywords etc.
anyhow, with `DeepDiff` you can set option `ignore_order=False` and have a direct comparison, but then that should be equivalent to `dict1 == dict2` 
I see if I can find reproducible examples.